plugins/cn0508.c

```diff
@@ -0,0 +1,323 @@
+
+static void entry_set_double(GtkWidget *entry, double value, int digits)
+{
+	gchar *s;
+
+	g_return_if_fail(GTK_IS_ENTRY(entry));
+	s = g_strdup_printf("%.*f", digits, value);
+	gtk_entry_set_text(GTK_ENTRY(entry), s);
+	g_free(s);
+}
+
+static double ad7124_voltage_conversion(long long raw, double vref, double gain)
+{
+	return ((raw / pow(2, N_BITS)) * vref / gain);
+}
+
+static int get_adc_voltage(struct iio_channel *adc_ch, double *out_data)
+{
+	long long raw;
+	int ret;
+
+	ret = iio_channel_attr_read_longlong(adc_ch, "raw", &raw);
+	if (!ret)
+		*out_data = ad7124_voltage_conversion(raw, V_REF_ADC, GAIN_ADC);
+
+	return ret;
+}
+
+static void cn0508_update_widgets(struct _cn0508_data *data)
+{
+	gchar text[10];
+
+	entry_set_double(u2_temp_monitor, data->u2_temp, 2);
+	entry_set_double(u3_temp_monitor, data->u3_temp, 2);
+	entry_set_double(dc_supply_current, data->output_current, 2);
+	entry_set_double(dc_supply_voltage, data->output_voltage, 2);
+	entry_set_double(in_voltage, data->input_voltage, 2);
+
+	snprintf(text, 5, "%4.2f", data->pot_voltage * 100);
+	strcat(text, "%");
+	gtk_progress_bar_set_fraction ((GtkProgressBar *)pot_voltage,
+				       data->pot_voltage);
+	gtk_progress_bar_set_text ((GtkProgressBar *)pot_voltage, text);
+
+	snprintf(text, 5, "%4.2f", data->pot_current * 100);
+	strcat(text, "%");
+	gtk_progress_bar_set_fraction ((GtkProgressBar *)pot_current,
+				       data->pot_current);
+	gtk_progress_bar_set_text ((GtkProgressBar *)pot_current, text);
+}
+
+static int cn0508_get_data(struct _cn0508_data *data)
+{
+	int ret;
+
+	ret = get_adc_voltage(u2_temp_ch, &data->adc_conversion);
+	if (ret)
+		return ret;
+	data->u2_temp = data->adc_conversion * 1000; /* 1mV/ºC */
+
+	ret = get_adc_voltage(u3_temp_ch, &data->adc_conversion);
+	if (ret)
+		return ret;
+	data->u3_temp = data->adc_conversion * 1000; /* 1mV/ºC */
+
+	ret = get_adc_voltage(out_current_ch, &data->adc_conversion);
+	if (ret)
+		return ret;
+	data->output_current = data->adc_conversion / 0.2; /* 200mV/A */
+
+	ret = get_adc_voltage(voltage_pot_pos_ch, &data->adc_conversion);
+	if (ret)
+		return ret;
+	data->pot_voltage = data->adc_conversion / V_REF_ADC; /* 0% = 0V, 100% = 2.5V */
+
+	ret = get_adc_voltage(current_pot_pos_ch, &data->adc_conversion);
+	if (ret)
+		return ret;
+	data->pot_current = data->adc_conversion / V_REF_ADC; /* 0% = 0V, 100% = 2.5V */
+
+	ret = get_adc_voltage(out_v_attenuator_ch, &data->adc_conversion);
+	if (ret)
+		return ret;
+	data->output_voltage = data->adc_conversion * 10.52;
+
+	ret = get_adc_voltage(in_v_attenuator_ch, &data->adc_conversion);
+	if (ret)
+		return ret;
+	data->input_voltage = data->adc_conversion * 14.33;
+
+	return ret;
+}
+
+static void update_values(void)
+{
+	iio_update_widgets(iio_widgets, num_widgets);
+}
+
+static void save_widget_value(GtkWidget *widget, struct iio_widget *iio_w)
+{
+	iio_w->save(iio_w);
+}
+
+static gboolean update_display(gpointer foo)
+{
+	int ret;
```

Comment:
doesn't the compiler warn about this maybe being un-initialized?

Reply:
seems it does:
https://travis-ci.org/analogdevicesinc/iio-oscilloscope/jobs/553747156

Reply:
yes it does, will fix.